dbms/src/Storages/KVStore/tests/gtest_new_kvstore.cpp

```diff
@@ -1164,4 +1164,80 @@ try
+try
+{
+    auto & ctx = TiFlashTestEnv::getGlobalContext();
+    ASSERT_NE(proxy_helper->sst_reader_interfaces.fn_key, nullptr);
+    UInt64 region_id = 1;
+    TableID table_id;
+
+    initStorages();
+    KVStore & kvs = getKVS();
+    table_id = proxy_instance->bootstrapTable(ctx, kvs, ctx.getTMTContext());
+    LOG_INFO(&Poco::Logger::get("Test"), "generated table_id {}", table_id);
+    proxy_instance->bootstrapWithRegion(kvs, ctx.getTMTContext(), region_id, std::nullopt);
+    auto kvr1 = kvs.getRegion(region_id);
+    auto r1 = proxy_instance->getRegion(region_id);
+    {
+        // Multiple files
+        MockSSTReader::getMockSSTData().clear();
+        MockSSTGenerator default_cf{902, 800, ColumnFamilyType::Default};
+        default_cf.insert(1, "v1");
+        default_cf.finish_file();
+        default_cf.insert(2, "v2");
+        default_cf.finish_file();
+        default_cf.insert(3, "v3");
+        default_cf.insert(4, "v4");
+        default_cf.finish_file();
+        default_cf.insert(5, "v5");
+        default_cf.insert(6, "v6");
+        default_cf.finish_file();
+        default_cf.insert(7, "v7");
+        default_cf.finish_file();
+        default_cf.freeze();
+        kvs.mutProxyHelperUnsafe()->sst_reader_interfaces = make_mock_sst_reader_interface();
+
+        auto make_meta = [&]() {
+            auto r2 = proxy_instance->getRegion(region_id);
+            auto modified_meta = r2->getState().region();
+            modified_meta.set_id(2);
+            modified_meta.set_start_key(RecordKVFormat::genKey(table_id, 1));
+            modified_meta.set_end_key(RecordKVFormat::genKey(table_id, 4));
+            modified_meta.add_peers()->set_id(2);
+            return modified_meta;
+        };
+        auto peer_id = kvr1->getMeta().peerId();
+        proxy_instance->debugAddRegions(
+            kvs,
+            ctx.getTMTContext(),
+            {2},
+            {{{RecordKVFormat::genKey(table_id, 0), RecordKVFormat::genKey(table_id, 4)}}});
+
+        // Overlap
+        EXPECT_THROW(
+            proxy_instance
+                ->snapshot(kvs, ctx.getTMTContext(), 2, {default_cf}, make_meta(), peer_id, 0, 0, std::nullopt, false),
+            Exception);
+
+        LOG_INFO(log, "Set to applying");
+        // region_state is "applying", but the key-range in proxy side still overlaps.
+        r1->mutState().set_state(raft_serverpb::PeerState::Applying);
+        ASSERT_EQ(proxy_helper->getRegionLocalState(1).state(), raft_serverpb::PeerState::Applying);
+        EXPECT_THROW(
+            proxy_instance
+                ->snapshot(kvs, ctx.getTMTContext(), 2, {default_cf}, make_meta(), peer_id, 0, 0, std::nullopt, false),
+            Exception);
+
+
+        LOG_INFO(log, "Shrink region 1");
+        // region_state is "applying", but the key-range in proxy side is not overlap.
+        r1->mutState().mutable_region()->set_start_key(RecordKVFormat::genKey(table_id, 0));
```

Comment:
```suggestion
        // region_state is "applying", but the key-range in proxy side is not overlap.
        r1->mutState().set_state(raft_serverpb::PeerState::Applying);
        r1->mutState().mutable_region()->set_start_key(RecordKVFormat::genKey(table_id, 0));
```
Add a line to explicitly set the state to applying here is more clear.